File: app/src/main/java/com/vitaliyhtc/dagger2investigation/presentation/productdetails/di/ProductDetailsModule.java
package com.vitaliyhtc.dagger2investigation.presentation.productdetails.di;

import com.vitaliyhtc.dagger2investigation.domain.ProductRepository;
import com.vitaliyhtc.dagger2investigation.presentation.productdetails.presenter.ProductDetailsPresenter;

import dagger.Module;
import dagger.Provides;

@Module
public class ProductDetailsModule {

    @Provides
    @ProductDetailsActivityScope
    ProductDetailsPresenter provideProductDetailsPresenter(ProductRepository productRepository) {
        return new ProductDetailsPresenter(productRepository);
    }
}

File: app/src/main/java/com/vitaliyhtc/dagger2investigation/presentation/productdetails/di/ProductDetailsModule.kt
package com.vitaliyhtc.dagger2investigation.presentation.productdetails.di

import com.vitaliyhtc.dagger2investigation.domain.ProductRepository
import com.vitaliyhtc.dagger2investigation.presentation.productdetails.presenter.ProductDetailsPresenter
import dagger.Module
import dagger.Provides

@Module
class ProductDetailsModule {

    @Provides
    @ProductDetailsActivityScope
    fun provideProductDetailsPresenter(productRepository: ProductRepository): ProductDetailsPresenter {
        return ProductDetailsPresenter(productRepository)
    }
}